harness/determined/common/experimental/model.py

```diff
@@ -74,6 +74,28 @@ def delete(self) -> None:
             self._session, modelName=self.model_name, modelVersionNum=self.model_version
         )
 
+    def get_metrics(self, group: str = "") -> Iterable["metrics.TrialMetrics"]:
+        """
+        Gets all metrics for a given metric group associated with this model version.
+        The checkpoint can be originally associated by calling
+        ``core_context.utils.report_task_using_model_version(<MODEL_VERSION>)``
+        from within a task.
+
+        Arguments:
+            group (str, optional): Group name for the metrics (example: "training", "validation").
+                All metrics will be returned when querying by "".
+        """
+
+        resp = bindings.get_GetTrialMetricsByModelVersion(
+            session=self._session,
+            modelName=self.model_name,
+            modelVersionNum=self.model_version,
+            trialSourceInfoType=bindings.v1TrialSourceInfoType.INFERENCE,
```

Comment:
I think it's a little surprising that we have normally associated metrics with checkpoints, but here we have ModelVersion.get_metrics() which does _not_ return the same concept of metrics that our checkpoints are normally associated with.

Can we rename this `get_inference_metrics` or something?

Reply:
I'm not actually sure what you mean by
> we have normally associated metrics with checkpoints
Do you mean within this PR? Or is there some other method to associate metrics with checkpoints that I'm not aware of?

Reply:
Also in regards to this particular `trialSourceInfoType`, it is called "inference" as opposed to "fine-tuning" for lore as we recognized some common functionality for this metric linking for their use case and added this field to allow for that distinction. We have not actually used that distinction yet, which is why this simply calls "inference" all the time. Perhaps that distinction shouldn't exist if we aren't using it yet? But I felt that it does no harm to be like that for now

Reply:
> Or is there some other method to associate metrics with checkpoints that I'm not aware of?

Yeah, the validation metrics that match it's `steps_completed` value.  They're exposed in the sdk [here](https://github.com/determined-ai/determined/blob/5c839013e43f06365a4a5a0af1a37c8d99f588de/harness/determined/common/experimental/checkpoint/_checkpoint.py#L58), which comes from the protobufs [here](https://github.com/determined-ai/determined/blob/5c839013e43f06365a4a5a0af1a37c8d99f588de/proto/src/determined/checkpoint/v1/checkpoint.proto#L44) which is pulled out the database [here](https://github.com/determined-ai/determined/blob/5c839013e43f06365a4a5a0af1a37c8d99f588de/master/static/migrations/20220504154053_add-checkpoints-v2-steps-completed-index.tx.up.sql#L39).

It is such a deeply-embedded association between checkpoints and metrics that that's where the checkpoint's searcher metric comes from in the webui, when you go to a trial and click on the "checkpoints" tab.

This feels like a new, quite different association between metrics and models (which are just annotated checkpoints), and that feels pretty confusing to me.